src/components/top-bar.tsx

```diff
@@ -32,40 +41,41 @@ export class TopBarComponent extends BaseComponent<IProps, IState> {
           </div>
           <div className="content-container right">
             <div className="select-mag">Add a 2nd magnet (movable)</div>
-            {this.renderBarButton(secondaryMagType, this.handleClickRightMagnetBarButton)}
-            {this.renderCoilButton(secondaryMagType, this.handleClickRightMagnetCoilButton)}
-            <div className="button" onClick={this.handleClickRightMagnetRemoveButton}>remove magnet</div>
+            {this.renderBarButton(secondaryMagType, secondaryDisabledClass, this.handleClickRightMagnetBarButton)}
+            {this.renderCoilButton(secondaryMagType, secondaryDisabledClass, this.handleClickRightMagnetCoilButton)}
+            <div className={"button text-only" + removeDisabledClass}
+                 onClick={this.handleClickRightMagnetRemoveButton}>Remove magnet</div>
           </div>
         </div>
-        {this.renderMagnetButton(secondaryMagType, "right", this.handleClickRightMagnetButton)}
+        {this.renderMagnetButton(secondaryMagText, secondaryMagType, "right", this.handleClickRightMagnetButton)}
       </div>
     );
   }
 
-  private renderMagnetButton = (magType: any, posClass: string, onClickFunction: () => void) => {
+  private renderMagnetButton = (text: string, magType: any, posClass: string, onClickFunction: () => void) => {
     const magImage = "assets/magnet-" + magType + "-icon.png";
     return (
       <div className={"magnet-button " + posClass} onClick={onClickFunction}>
       {magType === "none"
-        ? <div className={"unselected " + posClass}>select a magnet</div>
+        ? <div className={"unselected " + posClass}>{text}</div>
         : <img src={magImage} className={"icon " + posClass}/>}
       </div>
     );
   }
 
-  private renderBarButton = (magType: any, onClickFunction: () => void) => {
+  private renderBarButton = (magType: any, disabledClass: string, onClickFunction: () => void) => {
     return (
-      <div className="button short" onClick={onClickFunction}>
+      <div className={"button short" + disabledClass} onClick={onClickFunction}>
```

Comment:
This is a nit, but I'd probably include the space here rather than depending on it to get included in `disabledClass`

Reply:
Yes!  Much safer.

Reply:
Fixed.